### PR TITLE
feat: multiple config formats

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

[Description of the change]

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by JanDeDobbeleer** [comment]
@lnu @TravisTX when you have some time available, I'd love some additional manual validation on this one :-) I tested everything and "works on my machine" 😓

File: src/settings.go [hunk 1]
```diff
@@ -68,18 +74,56 @@ func loadUserConfiguration(env environmentInfo) (*Settings, error) {
...
 	return &settings, nil
 }
 
+func exportSettings(settingsFile, format string) string {
+	if len(format) == 0 {
+		format = config.JSON
+	}
+	err := config.LoadFiles(settingsFile)
+	if err != nil {
+		return "UNABLE TO OPEN CONFIG"
```
Comment (tradiff) [inline src/settings.go]: I'm hitting this if I have invalid json, yaml, or toml.  What about an error more like "INVALID CONFIG" in that situation?  For example, if I use a trailing comma in the last element of a json block, or indent too much in yaml, or add a stray `]` in toml.  I also noticed `err` says exactly what the problem was and what line it's on, which could be helpful to the user.  Maybe just `fmt.Println` it?  Just an idea.

Comment (JanDeDobbeleer) [reply src/settings.go]: I'll add the context indeed.

**Comment by tradiff** [comment]
Everything seems to be working correctly. I can't figure out how to get the $schema to work in yaml using vscode.  My understanding is the json schema should also work with yaml.  I do have the `vscode-yaml` extension, which gives me coloring, but seems to not be giving me the context from the schema.  Have you had any luck with this?

**Comment by lnu** [comment]
Will give a try tomorrow morning. Good work 👍

**Comment by JanDeDobbeleer** [comment]
Vscode autocomplete is on my list. Didn't look at it yet.

**Comment by lnu** [comment]
image

~json can be removed~ => "in the default output format(json)" or something

**Comment by lnu** [comment]
> Everything seems to be working correctly. I can't figure out how to get the $schema to work in yaml using vscode. My understanding is the json schema should also work with yaml. I do have the `vscode-yaml` extension, which gives me coloring, but seems to not be giving me the context from the schema. Have you had any luck with this?

It works here when setting the schema url in user config. But it fails when set at the top of the yaml file(it should work):
`# yaml-language-server: $schema=<urlToTheSchema>`


=> works with the right syntax `# yaml-language-server: $schema=https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json`

We can add this extension `redhat.vscode-yaml` to extensions.json in .vscode(support for yaml autocomplete)

File: src/settings.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/settings.go]: done

Comment (lnu) [reply src/settings.go]: works fine 👍

**Comment by lnu** [comment]
The only downside of yaml and toml is the lack of inline schema definition like the json one. ~Should work with yaml but somewhat fails and~ toml supports only schema uploaded to a central repo.
For the pull, looks good to me.

**Comment by lnu** [comment]
even better toml extension supports toml schema but not inline. https://github.com/tamasfe/taplo/issues/95.

**Comment by JanDeDobbeleer** [comment]
Ok. So, on my to-do list:
- Add inline schemas for supported languages if it's not present
- Create helper functions to avoid adding your current config (like pwsh does)

**Comment by lnu** [comment]
Tested all your changes(export_poshconfig and export-poshtheme) and it works well 👍

**Comment by JanDeDobbeleer** [comment]
@lnu maybe I should also align the name, no? `Export-PoshConfig` makes sense.

**Pull request merged**